pkg/common/security/usergroup_test.go

```diff
@@ -251,4 +358,14 @@ func TestConvertUGI(t *testing.T) {
 	if err == nil {
 		t.Errorf("invalid username, convert should have failed: %v", err)
 	}
+
+	// try unknown user with empty group when forced
+	ugi.User = "unknown"
+	ugi.Groups = []string{}
+	ug, err = testCache.ConvertUGI(ugi, true)
+	exceptedGroup := []string{common.AnonymousGroup}
+	testCache.lock.Lock()
+	assert.Assert(t, reflect.DeepEqual(ug.Groups, exceptedGroup), "group should be: %v, but got: %v", exceptedGroup, ug.Groups)
+	testCache.lock.Unlock()
```

Comment:
This lock/unlock pair can be removed. `ConvertUGI()` returns a copy and internally uses a mutex to access the map.